core/Array.carp

```diff
@@ -240,4 +240,18 @@
     (zip &Pair.init-from-refs
          &(range 0 (length xs) 1) ;; Inefficient, creates a temporary array.
          xs))
+
+  (doc remove "Remove occurrences of element el in array arr, in-place")
```

Comment:
I’m not familiar with our filter, Is our filter really an inplace mutation?
what does In place mean here?

Usually our functions which mutate things in place end with an exclamation mark ala scheme, at least within the array libraries. Is that convention no more? (I haven’t been around for a bit).

Reply:
It is, but it’s save because it takes ownership, and thus we do not put a `!` at the end. That’s why I opened #317 (I want a version that doesn’t take ownership as well).

Reply:
Yeah, we should resolve that (it's basically just naming decisions left to do there, IIRC). I'll merge this for now, since this looks useful on its own.